internal/server/admin/certificate_service.go

```diff
@@ -222,7 +226,26 @@ func (s *CertificateService) logger(ctx context.Context) *zap.Logger {
 	return util.LoggerFromContext(ctx).With(s.loggerFields...)
 }
 
-func certificatesFromObjects(objects []model.Object) ([]*pb.Certificate, error) {
+func addSNIMetadata(ctx context.Context, cert *pb.Certificate, db store.Store) error {
+	snis, err := getAllSNIsForCertificate(ctx, cert.Id, db)
+	if err != nil {
+		return fmt.Errorf(
+			"retrieving snis used by certificate '%s': %w",
+			cert.Id, err,
+		)
+	}
+	if len(snis) > 0 && cert.Metadata == nil {
+		cert.Metadata = &pb.CertificateMetadata{}
+	}
+	for _, sni := range snis {
+		cert.Metadata.Snis = append(cert.Metadata.Snis, sni.Name)
+	}
```

Comment:
nit: This can just be:
```suggestion
	if len(snis) > 0 && cert.Metadata == nil {
		cert.Metadata = &pb.CertificateMetadata{Snis: make([]string, len(snis))}
	}
	for i, sni := range snis {
		cert.Metadata.Snis[i] = sni.Name
	}
```